packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/Timestamp.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal.interop
+
+/**
+ * Wrapper around Core Timestamp values.
+ * See https://github.com/realm/realm-core/blob/master/src/realm/timestamp.hpp for more information
+ */
+open class Timestamp(open val seconds: Long, open val nanoSeconds: Int)
```

Comment:
I guess we can't just have this as an interface as I said since we also need to instantiate these in `cinterop`. 
But with an `cinterop`-interface you could let the `library` implementation implement that and skip one of the translations. But probably cleaner to keep them completely separate and follow the typemapping thoughts.

Reply:
Yeah, this was turning into a bit of a mess.....It felt a bit much having both `RealmInstant`, `RealmInstantImpl`, `Timestamp` and `TimestampImpl`....Having `RealmInstant` being a class with an inner Timestamp was the smallest I could make the type system, but that went a bit against our decision to keep the API as interfaces + made problems when moving the type to Cinterop. 

For now I would probably prefer to keep it like this, and then we can iterate once we get more types and/or start looking into type converters.

Reply:
After thinking a bit more...i ended up converting Timestamp to an interface, this allows both RealmInstantImpl and TimestampImpl to be data classes.